app/grandchallenge/participants/forms.py

```diff
@@ -1,8 +1,110 @@
-from django.forms import HiddenInput, ModelForm, TextInput
+from django.forms import (
+    HiddenInput,
+    ModelForm,
+    TextInput,
+    inlineformset_factory,
+)
 
 from grandchallenge.core.forms import SaveFormInitMixin
 from grandchallenge.core.widgets import JSONEditorWidget
-from grandchallenge.participants.models import RegistrationQuestion
+from grandchallenge.participants.form_fields import (
+    RegistrationQuestionAnswerField,
+)
+from grandchallenge.participants.models import (
+    RegistrationQuestion,
+    RegistrationQuestionAnswer,
+    RegistrationRequest,
+)
+
+
+class RegistrationRequestForm(ModelForm):
+
+    class Meta:
+        model = RegistrationRequest
+        fields = (
+            "user",
+            "challenge",
+        )
+        widgets = {
+            "user": HiddenInput(),
+            "challenge": HiddenInput(),
+        }
+
+    def __init__(self, *args, challenge, user, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user"].initial = user
+        self.fields["user"].disabled = True
+        self.fields["challenge"].initial = challenge
+        self.fields["challenge"].disabled = True
+
+        self.answer_formset = self._get_answer_formset(
+            challenge,
+            data=kwargs.get("data"),
+        )
+
+    def _get_answer_formset(self, challenge, data):
+        questions = challenge.registration_questions.all()
+        answer_formset_factory = inlineformset_factory(
+            parent_model=self._meta.model,
+            model=RegistrationQuestionAnswer,
+            form=RegistrationQuestionAnswerForm,
+            max_num=questions.count(),
+            min_num=questions.count(),
+            absolute_max=questions.count(),
+            extra=0,
+            can_delete=False,
+            can_delete_extra=False,
+        )
+        return answer_formset_factory(
+            instance=self.instance,
+            data=data,
+            initial=[{"question": q} for q in questions],
+        )
+
+    def is_valid(self):
+        return super().is_valid() and self.answer_formset.is_valid()
+
+    def full_clean(self, *args, **kwargs):
+        super().full_clean(*args, **kwargs)
+        self.answer_formset.full_clean()
+
+        # Show formset-level validation errors
+        for error in self.answer_formset.non_form_errors():
+            self.add_error(field=None, error=error)
+
+    def save(self, *args, **kwargs):
+        registration_request = super().save(*args, **kwargs)
+
+        # By default, formsets only save changed or non-empty
+        # forms. Short-circuit saving to ensure it saves empty
+        # answers
+        for form in self.answer_formset.forms:
+            self.answer_formset.save_new(form)
```

Comment:
This seems a bit iffy. Doesn't Django skip validation for empty forms? Can you force that validation and for the formset to be considered? Can you set `empty_permitted` on the form instead?

Reply:
Only if `empty_permitted=True` is set, are empty forms not validated. The `BaseForm` has this in its `full_clean`:
```Python
if self.empty_permitted and not self.has_changed():
   return

self._clean_fields()
self._clean_form()
self._post_clean()
```

This is the construct that is being short-circuited, which is called from the formset' `save()`:

```Python
def save_new_objects(self, commit=True):
    self.new_objects = []
    for form in self.extra_forms:
        if not form.has_changed():
            continue
        # If someone has marked an add form for deletion, don't save the
        # object.
        if self.can_delete and self._should_delete_form(form):
            continue
        self.new_objects.append(self.save_new(form, commit=commit))
        if not commit:
            self.saved_forms.append(form)
    return self.new_objects
```

The only other entry to this would be to:
- Short circuit `has_changed()` in the form.
- Not commit
  - Save the `saved_forms`
  - Save the `extra_forms` not in `saved_forms`, via `save_new()`

The first is a clear hack. The latter is equivalent to the current: there are only extra forms.